Scripts/extras/pull_and_build_from_git.py

```diff
@@ -803,15 +828,31 @@ def prepare_build(platform_name, base_folder, build_folder, package_root_folder,
                                target_platform_name=platform_name,
                                target_platform_config=target_platform_config)
 
-    # Validate the cmake find template
-    if not package_info.cmake_find_template:
-        raise BuildError("Missing 'cmake_find_template' entry in build config")
-    if os.path.isabs(package_info.cmake_find_template):
-        cmake_find_template_path = pathlib.Path(package_info.cmake_find_template)
+    cmake_find_template_path = None
+    cmake_find_source_path = None
+
+    if package_info.cmake_find_template is not None:
+
+        # Validate the cmake find template
+        if os.path.isabs(package_info.cmake_find_template):
+            cmake_find_template_path = pathlib.Path(package_info.cmake_find_template)
+        else:
+            cmake_find_template_path = base_folder_path / package_info.cmake_find_template
+        if not cmake_find_template_path.is_file():
+            raise BuildError("Invalid 'cmake_find_template' entry in build config")
+
+    elif package_info.cmake_find_source is not None:
+
+        # Validate the cmake find source
+        if os.path.isabs(package_info.cmake_find_source):
+            cmake_find_source_path = pathlib.Path(package_info.cmake_find_source)
+        else:
+            cmake_find_source_path = base_folder_path / package_info.cmake_find_source
```

Comment:
There is no reason to perform an isabs check, the pathlib library will replace previous components if it finds an absolute path component when appended paths: https://docs.python.org/3/library/os.path.html#os.path.join
> If a component is an absolute path, all previous components are thrown away and joining continues from the absolute path component.

Therefore this can be changed to just be
```suggestion
        # Validate the cmake find source
        cmake_find_source_path = base_folder_path / package_info.cmake_find_source
```

Reply:
Actually, we shouldn't even allow absolute paths here. I'll add an abs check here anyways as an assert